homeassistant/components/nfandroidtv/config_flow.py

```diff
@@ -4,21 +4,48 @@
 import logging
 from typing import Any
 
-from notifications_android_tv.notifications import ConnectError, Notifications
+from notifications_android_tv.notifications import (
+    ConnectError,
+    Notifications,
+    test_port,
+)
 import voluptuous as vol
 
 from homeassistant import config_entries
+from homeassistant.components.dhcp import DhcpServiceInfo
 from homeassistant.const import CONF_HOST, CONF_NAME
 from homeassistant.data_entry_flow import FlowResult
+from homeassistant.helpers.device_registry import format_mac
 
-from .const import DEFAULT_NAME, DOMAIN
+from .const import ANDROID_TV_NAME, DEFAULT_NAME, DOMAIN, FIRE_TV_NAME, PLACEHOLDERS
 
 _LOGGER = logging.getLogger(__name__)
 
 
 class NFAndroidTVFlowHandler(config_entries.ConfigFlow, domain=DOMAIN):
     """Handle a config flow for NFAndroidTV."""
 
+    def __init__(self) -> None:
+        """Initialize an NFAndroidTV flow."""
+        self.ip_address = ""
+
+    async def async_step_dhcp(self, discovery_info: DhcpServiceInfo) -> FlowResult:
+        """Handle dhcp discovery."""
+        self.ip_address = discovery_info.ip
+        mac = format_mac(discovery_info.macaddress)
+
+        # Host unique_id has been deprecated in 2022.6
+        if existing_entry := await self.async_set_unique_id(discovery_info.ip):
+            self.hass.config_entries.async_update_entry(existing_entry, unique_id=mac)
+            return self.async_abort(reason="already_configured")
```

Comment:
We should enumerate current entries and look for CONF_HOST in data instead and set it instead